### PR TITLE
DM-21043: Qserv log diet: use named context for query ID

### PR DESCRIPTION
**Pull request by andy-slac**
Switch to using MDC for some context information instead of adding explicit info to each message.

### CODE REVIEW

File: core/modules/czar/Czar.cc [hunk 1]
```diff
@@ -134,6 +136,9 @@ Czar::submitQuery(std::string const& query,
     std::string const msgTableName = "message_" + userQueryId;
     std::string const lockName = resultDb + "." + msgTableName;
 
+    // Add logging context with user query ID
+    LOG_MDC_SCOPE("TID", userQueryId);
```
Comment (ktlim) [inline core/modules/czar/Czar.cc]: Curious: How is TID different from QID?

**Comment by andy-slac** [comment]
@jgates108, travis is failing, this is because it does not know how to pull new `log` version.

File: core/modules/czar/Czar.cc (hunk 1 above)
Comment (andy-slac) [reply core/modules/czar/Czar.cc]: We just had 1-hour discussion yesterday about how we want to get rid of TID 🙂 TID is a sort of temporary ID which is assigned to a request before the request is registered in QMeta and gets it actual QID. Some requests never go to QMeta, so for those we only have TID. The whole things is a mess and we'd like to get rid of TID, but it's not quite trivial.

Comment (andy-slac) [reply core/modules/czar/Czar.cc]: Just to add - TID is a sort of randomly generated but not unique and not persistent, QID is unique and persistent.

**Pull request merged**